chall03/krusthol.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func log_prefixed(msg string) {
+	fmt.Printf("%d", time.Since(StartTime).Milliseconds())
+	log.Println("ms - ", msg)
+}
+
+func get_request(address string) (id_rgb_str string) {
+	log_prefixed("GET " + address)
+	chall_req, err := http.Get(address)
+	if err != nil {
+		log.Fatal(err)
+	}
+	raw_id_rgb, err := ioutil.ReadAll(chall_req.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	chall_req.Body.Close()
+	id_rgb_str = fmt.Sprintf("%s", raw_id_rgb)
+	log_prefixed("GET " + address + "\n\tanswer: " + id_rgb_str)
+	return
+}
+
+func regexp_parse_before_slash(chall_body string) string {
+	halved_string := strings.Split(chall_body, "-")
+	reg, err := regexp.Compile("[^0-9=]+")
```

Comment:
For this exercise I would recommend using Sscanf instead of regex: https://softwareengineering.stackexchange.com/a/113243